parlai/core/torch_ranker_agent.py

```diff
@@ -313,9 +318,29 @@ def eval_step(self, batch):
             cand_preds_generator = (cand_list[rank] for rank in ordering)
             cand_preds.append(list(islice(cand_preds_generator, max_preds)))
 
+        if (self.opt.get('repeat_blocking_heuristic', False) and
```

Comment:
why do we do false here if it defaults to true above?

Reply:
if the option isn't there this is backwards compatible to behavior before i guess.. but ok i can change it

Reply:
so you're ok with this not being backwards compatible, right?

Reply:
i don't think this will effect anything because if the argument didn't exist beforehand it will default to True anyway

Reply:
oh... what code does that?

Reply:
The parser will insert the default I think.

Reply:
yep i believe that is correct

Reply:
so in general this is bad to set a new default which isn't backwards compatible, but done anyway. i wonder if we could have a     default_if_missing=False     in the  agent.add_argument function?

Reply:
Yeah this is kinda all the domain of `upgrade_opt` that we've been debating.

Reply:
yes, but this suggestion wouldn't require to write a new function for this standard use-case